src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/service/YFlowUpdateService.java

```diff
@@ -0,0 +1,319 @@
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.flowhs.service;
+
+import org.openkilda.floodlight.api.response.SpeakerResponse;
+import org.openkilda.floodlight.flow.response.FlowErrorResponse;
+import org.openkilda.messaging.command.yflow.SubFlowDto;
+import org.openkilda.messaging.command.yflow.SubFlowPatchDto;
+import org.openkilda.messaging.command.yflow.SubFlowSharedEndpointEncapsulation;
+import org.openkilda.messaging.command.yflow.YFlowPatchRequest;
+import org.openkilda.messaging.command.yflow.YFlowRequest;
+import org.openkilda.messaging.command.yflow.YFlowRequest.Type;
+import org.openkilda.messaging.error.ErrorType;
+import org.openkilda.model.FlowEndpoint;
+import org.openkilda.model.SwitchId;
+import org.openkilda.model.YFlow;
+import org.openkilda.pce.PathComputer;
+import org.openkilda.persistence.PersistenceManager;
+import org.openkilda.persistence.repositories.RepositoryFactory;
+import org.openkilda.persistence.repositories.YFlowRepository;
+import org.openkilda.wfm.CommandContext;
+import org.openkilda.wfm.share.flow.resources.FlowResourcesManager;
+import org.openkilda.wfm.share.utils.FsmExecutor;
+import org.openkilda.wfm.topology.flowhs.exception.DuplicateKeyException;
+import org.openkilda.wfm.topology.flowhs.exception.UnknownKeyException;
+import org.openkilda.wfm.topology.flowhs.fsm.yflow.update.YFlowUpdateContext;
+import org.openkilda.wfm.topology.flowhs.fsm.yflow.update.YFlowUpdateFsm;
+import org.openkilda.wfm.topology.flowhs.fsm.yflow.update.YFlowUpdateFsm.Event;
+import org.openkilda.wfm.topology.flowhs.mapper.YFlowRequestMapper;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Slf4j
+public class YFlowUpdateService
+        extends FsmBasedFlowProcessingService<YFlowUpdateFsm, Event, YFlowUpdateContext, YFlowUpdateHubCarrier> {
+    private final YFlowUpdateFsm.Factory fsmFactory;
+    private final YFlowRepository yFlowRepository;
+
+    public YFlowUpdateService(YFlowUpdateHubCarrier carrier, PersistenceManager persistenceManager,
+                              PathComputer pathComputer, FlowResourcesManager flowResourcesManager,
+                              FlowUpdateService flowUpdateService,
+                              int resourceAllocationRetriesLimit, int speakerCommandRetriesLimit) {
+        super(new FsmExecutor<>(Event.NEXT), carrier, persistenceManager);
+        fsmFactory = new YFlowUpdateFsm.Factory(carrier, persistenceManager, pathComputer,
+                flowResourcesManager, flowUpdateService, resourceAllocationRetriesLimit, speakerCommandRetriesLimit);
+        RepositoryFactory repositoryFactory = persistenceManager.getRepositoryFactory();
+        yFlowRepository = repositoryFactory.createYFlowRepository();
+
+        flowUpdateService.addEventListener(new FlowUpdateEventListener() {
```

Comment:
Why not to define "normal" nested class, instead of this anonymous class?

Reply:
I don't think that makes sense. The definition of this class is moved to a separate method.